File: internal/repository/user_settings.py
from sqlalchemy import Column, String, Boolean, TIMESTAMP, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.sql import func
import uuid

Base = declarative_base()

class UserSettings(Base):
    __tablename__ = 'user_settings'
    
    user_settings_id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_id = Column(UUID(as_uuid=True), ForeignKey('users.id'), nullable=False, unique=True)
    
    notifications_enabled = Column(Boolean, nullable=False, default=True)
    language = Column(String(5), nullable=False, default='ru')
    timezone = Column(String(50), nullable=False, default='Europe/Moscow')
    privacy_level = Column(String(20), nullable=False, default='public')
    
    updated_at = Column(TIMESTAMP, server_default=func.now(), onupdate=func.now())

File: service/internal/domain/userSettings.go
package model

import (
    "time"
    "github.com/google/uuid"
)

type UserSettings struct {
    SettingsID         uuid.UUID    `json:"settings_id" db:"settings_id"`
    UserID             uuid.UUID    `json:"user_id" db:"user_id"`
    NotificationsEnabled bool       `json:"notifications_enabled" db:"notifications_enabled"`
    Language           string       `json:"language" db:"language"`
    Timezone           string       `json:"timezone" db:"timezone"`
    PrivacyLevel       PrivacyLevel `json:"privacy_level" db:"privacy_level"`
    UpdatedAt          time.Time    `json:"updated_at" db:"updated_at"`
}

type PrivacyLevel string

const (
    PrivacyLevelPublic  PrivacyLevel = "PUBLIC"
    PrivacyLevelPrivate PrivacyLevel = "PRIVATE"
    PrivacyLevelFriends PrivacyLevel = "FRIENDS"
    PrivacyLevelOnlyMe  PrivacyLevel = "ONLY_ME"
)